dotCMS/src/main/java/com/dotmarketing/portlets/workflows/business/WorkflowFactoryImpl.java

```diff
@@ -436,32 +437,37 @@ public WorkflowScheme findScheme(String id) throws DotDataException {
 		return scheme;
 	}
 
-	public WorkflowScheme findSchemeForStruct(String structId) throws DotDataException {
-
+	public List<WorkflowScheme> findSchemesForStruct(String structId) throws DotDataException {
+		List<WorkflowScheme> schemes = new ArrayList<>();
 		if (LicenseUtil.getLevel() < LicenseLevel.STANDARD.level) {
-			return this.findDefaultScheme();
+			schemes.add(this.findDefaultScheme());
+			return schemes;
 		}
 
-		WorkflowScheme scheme = null;
-		scheme = cache.getSchemeByStruct(structId);
+		schemes = cache.getSchemesByStruct(structId);
 
-		if (scheme != null) {
-			return scheme;
+		if (schemes != null && !schemes.isEmpty()) {
+			return schemes;
 		}
 
 		final DotConnect db = new DotConnect();
 		db.setSQL(sql.SELECT_SCHEME_BY_STRUCT);
 		db.addParam(structId);
 		try {
-			scheme = (WorkflowScheme) this.convertListToObjects(db.loadObjectResults(), WorkflowScheme.class).get(0);
+			schemes = this.convertListToObjects(db.loadObjectResults(), WorkflowScheme.class);
+			if(schemes.isEmpty()){
+				schemes.add(this.findDefaultScheme());
+			}
 		} catch (final Exception er) {
-
-			scheme = this.findDefaultScheme();
-
+			try {
+				schemes.add(this.findDefaultScheme());
+			}catch(Exception ex){

```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Catch a list of specific exception subtypes instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2221)